### PR TITLE
[Chore/#40] msw를 이용한 프로필 GET 요청 테스트

### PR DESCRIPTION
**Pull request by SebellKo**
## 📌 작업 내용

msw를 이용해서 프로필 GET 요청을 테스트 했습니다.

<br/>

## 📌 구현 결과 (선택)

```javascript
import { http, HttpResponse } from 'msw';

const storeProfile = {
  business_name: '우리집',
  contact: '010-1111-1111',
  address: '대전광역시 동구 삼성동',
  img: 'https://...',
  sector_name: '한식',
};

const companyProfile = {
  business_name: '업체명',
  contact: '010-1111-1111',
  img: 'https://...',
  address: '대전광역시 동구 삼성동',
  sector_name: '도매업체',
  homepage: 'https://...',
  introduce: '도매업체 소개',
};

export const profileHandler = [
  http.get('/profile/:id?type="shop"', async () => {
    return HttpResponse.json(storeProfile);
  }),

  http.get('/profile/:id?type="company"', async () => {
    return HttpResponse.json(companyProfile);
  }),

  http.put('/users/profile', async ({ request }) => {
    const updateProfile = await request.json();
    console.log('Update Profile: ', updateProfile);
  }),
];
```

스크린샷 2024-08-05 오후 6 59 14

[ **_msw를 이용한 데이터로 각 placeholder들이 교체된 모습_** ]

<br/>

## 📌 기타 사항

사용방법은 아래 설명과 같습니다.

1. src/mocks 폴더 내부 아래 사진과 같이 테스트할 파일을 생성한다.
 
스크린샷 2024-08-05 오후 7 01 23

2. 아래와 같이 코드를 작성한다.

- `http` 메소드를 사용해 요청을 보낼 수 있습니다.
- `HttpResponse` 메소드를 사용해 응답을 보낼 수 있습니다.
- param을 가져올때는 `const {id} = request.params` 이렇게 사용하면 됩니다.
- 쿼리 스트링을 가져올때는 아래와 같이 사용하면 됩니다.

```javascript
http.get('/product', ({ request }) => {
    const url = new URL(request.url)

    const productId = url.searchParams.get('id')
```

```javascript
import { http, HttpResponse } from 'msw';

const storeProfile = {
  business_name: '우리집',
  contact: '010-1111-1111',
  address: '대전광역시 동구 삼성동',
  img: 'https://...',
  sector_name: '한식',
};

const companyProfile = {
  business_name: '업체명',
  contact: '010-1111-1111',
  img: 'https://...',
  address: '대전광역시 동구 삼성동',
  sector_name: '도매업체',
  homepage: 'https://...',
  introduce: '도매업체 소개',
};

export const profileHandler = [
  http.get('/profile/:id?type="shop"', async () => {
    return HttpResponse.json(storeProfile);
  }),

  http.get('/profile/:id?type="company"', async () => {
    return HttpResponse.json(companyProfile);
  }),

  http.put('/users/profile', async ({ request }) => {
    const updateProfile = await request.json();
    console.log('Update Profile: ', updateProfile);
  }),
];
```

3. src/mocks/worker.ts 파일 내부 `handlers` 배열에 스프레드 연산자를 이용해 넣어주면 됩니다.

```javascript
import { setupWorker } from 'msw/browser';
import { profileHandler } from './profile';

const handlers = [...profileHandler, ...newHandler];

export const worker = setupWorker(...handlers);
```

위와 같이 설정했다면 msw가 서버로 보내는 요청을 인터셉터해 만들어둔 목데이터로 처리를 해줍니다.

감사합니다.

<br/>

### CODE REVIEW

**Review by JIMIN1020 — APPROVED** [review]

**Review by juhee067 — APPROVED** [review]
굳~!

**Comment by JIMIN1020** [comment]
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">0%</td> <td align="right">0 / 3687</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">0%</td> <td align="right">0 / 3687</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">1.41%</td> <td align="right">2 / 141</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">1.41%</td> <td align="right">2 / 141</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">% Branch</th> <th align="right">% Funcs</th> <th align="right">% Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/public/mockServiceWorker.js">public/mockServiceWorker.js</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/public/mockServiceWorker.js#L1-L284" class="text-red">1-284</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/main.tsx">src/main.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/main.tsx#L1-L4" class="text-red">1-4</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/main.tsx#L6" class="text-red">6</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/main.tsx#L8-L23" class="text-red">8-23</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/main.tsx#L25" class="text-red">25</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/getProfile.ts">src/features/Profile/api/getProfile.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/getProfile.ts#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/getProfile.ts#L3-L6" class="text-red">3-6</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/putProfile.ts">src/features/Profile/api/putProfile.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/putProfile.ts#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/api/putProfile.ts#L4-L7" class="text-red">4-7</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx">src/features/Profile/components/EditProfile/CategoryButtons.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L1-L5" class="text-red">1-5</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L7-L19" class="text-red">7-19</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L21-L22" class="text-red">21-22</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L24-L26" class="text-red">24-26</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L28" class="text-red">28</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L30" class="text-red">30</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L36" class="text-red">36</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L42" class="text-red">42</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx">src/features/Profile/components/EditProfile/CompanyProfile.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L1-L8" class="text-red">1-8</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L10-L12" class="text-red">10-12</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L14-L18" class="text-red">14-18</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L20-L46" class="text-red">20-46</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L48" class="text-red">48</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L50-L51" class="text-red">50-51</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L53" class="text-red">53</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L55" class="text-red">55</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L63" class="text-red">63</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L69" class="text-red">69</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L74" class="text-red">74</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L79-L80" class="text-red">79-80</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/CompanyProfile.tsx#L83" class="text-red">83</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx">src/features/Profile/components/EditProfile/StoreProfile.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L3-L7" class="text-red">3-7</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L9-L10" class="text-red">9-10</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L12" class="text-red">12</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L14-L30" class="text-red">14-30</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L32-L35" class="text-red">32-35</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L37" class="text-red">37</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L39" class="text-red">39</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L43" class="text-red">43</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L49-L50" class="text-red">49-50</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L56" class="text-red">56</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/components/EditProfile/StoreProfile.tsx#L60" class="text-red">60</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileMutation.ts">src/features/Profile/hooks/useProfileMutation.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileMutation.ts#L1-L2" class="text-red">1-2</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileMutation.ts#L5-L13" class="text-red">5-13</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileMutation.ts#L15" class="text-red">15</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileQuery.ts">src/features/Profile/hooks/useProfileQuery.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileQuery.ts#L1-L2" class="text-red">1-2</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileQuery.ts#L5-L9" class="text-red">5-9</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileQuery.ts#L11-L12" class="text-red">11-12</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/hooks/useProfileQuery.ts#L14" class="text-red">14</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/features/Profile/model/profile.model.ts">src/features/Profile/model/profile.model.ts</a></td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="left"></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts">src/mocks/profile.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L3-L9" class="text-red">3-9</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L11-L19" class="text-red">11-19</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L21-L24" class="text-red">21-24</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L26-L28" class="text-red">26-28</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/profile.ts#L30-L34" class="text-red">30-34</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/worker.ts">src/mocks/worker.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/e1ba6b3a690ee22cdf88d78bed20700e943493ba/src/mocks/worker.ts#L1-L6" class="text-red">1-6</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/Licruit/Licruit-frontend/actions/runs/10247311479>#56</a></em>

File: src/features/Profile/api/getProfile.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { httpClient } from '@/api/http';
+
+export const getProfile = async (id: string, type: string) => {
+  const result = await httpClient.get(`/profile/${id}?type=${type}`);
```
Comment (ksy90101) [inline src/features/Profile/api/getProfile.ts:4]: type이 필수값인가요?

File: src/features/Profile/api/getProfile.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  return result.data;
```
Comment (ksy90101) [inline src/features/Profile/api/getProfile.ts:5]: 이거에 대한 Type이 없네요.

File: src/features/Profile/api/getProfile.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { httpClient } from '@/api/http';
+
+export const getProfile = async (id: string, type: string) => {
```
Comment (ksy90101) [inline src/features/Profile/api/getProfile.ts:3]: id는 number 아닌가요?
type은 따로 안정해져있나요? 단순히 string이 아닐꺼 같은데.

File: src/features/Profile/api/putProfile.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { httpClient } from '@/api/http';
+import { Profile } from '../model/profile.model';
+
+export const putProfile = async (data: Profile) => {
```
Comment (ksy90101) [inline src/features/Profile/api/putProfile.ts:4]: 여기도 응답 타입이 명시되어 있지 않네요.

File: src/features/Profile/components/EditProfile/CompanyProfile.tsx [hunk 5]
```diff
@@ -1,35 +1,46 @@
...
-        placeholder='최근학'
+        placeholder={profile.business_name}
         isRequired
         hasValidation
       />
       <IntroduceWrapper>
         <Label label={LABEL.introduce} />
-        <Introduce placeholder='내용을 입력해주세요' maxLength={400} />
+        <Introduce
+          placeholder={profile.introduce || '내용을 입력해주세요'}
```
Comment (ksy90101) [inline src/features/Profile/components/EditProfile/CompanyProfile.tsx:31]: placeholder가 profile.introduce맞나요?
이상한데요...
실제 profile의 introduce가 있다면 그게 값이되어야 하지 않나요?

File: src/features/Profile/hooks/useProfileMutation.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { useMutation } from '@tanstack/react-query';
+import { putProfile } from '../api/putProfile';
+import { Profile } from '../model/profile.model';
+
+const useProfileMutation = () => {
+  return useMutation({
+    mutationFn: (data: Profile) => putProfile(data),
+    onError: (error) => {
+      console.log(error);
```
Comment (ksy90101) [inline src/features/Profile/hooks/useProfileMutation.ts:9]: Lint 잘 보고. ㅓ지하셨으면 좋겠어요.

File: src/features/Profile/hooks/useProfileQuery.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@tanstack/react-query';
+import { getProfile } from '../api/getProfile';
+import { Profile } from '../model/profile.model';
+
+const useProfileQuery = (id: string, type: string) => {
```
Comment (ksy90101) [inline src/features/Profile/hooks/useProfileQuery.ts:5]: 여기도 id가 strign...

File: src/mocks/profile.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    return HttpResponse.json(storeProfile);
+  }),
+
+  http.get('/profile/:id?type="company"', async () => {
+    return HttpResponse.json(companyProfile);
+  }),
+
+  http.put('/users/profile', async ({ request }) => {
+    const updateProfile = await request.json();
+    console.log('Update Profile: ', updateProfile);
```
Comment (ksy90101) [inline src/mocks/profile.ts:32]: 여기도 린트;;

File: src/features/Profile/api/getProfile.ts (hunk 1 above)
Comment (SebellKo) [reply src/features/Profile/api/getProfile.ts:4]: 사용자가 도매업체인지 소상공인인지에 따른 보여주는 프로필 수정 컴포넌트가 달라서 요청에 포함 시켰습니다 !!
아직 백엔드에서는 미구현인 상태라 백엔드와 얘기 나눠봐야 할것 같습니다 !!

File: src/features/Profile/api/getProfile.ts (hunk 3 above)
Comment (SebellKo) [reply src/features/Profile/api/getProfile.ts:3]: 매개변수로 넘어올때 string값으로 넘어오기 때문에 string으로 설정했습니다 !
타입설정은 변수의 의미에 맞춰서 설정해야 하는게 좋을까요 ?!

제가 타입스크립트에 숙련도가 높지않아 다시 여쭤봅니다 !!

File: src/features/Profile/api/putProfile.ts (hunk 4 above)
Comment (SebellKo) [reply src/features/Profile/api/putProfile.ts:4]: 응답타입 학습해보겠습니다 !!
감사합니다 !!!

File: src/features/Profile/components/EditProfile/CompanyProfile.tsx (hunk 5 above)
Comment (SebellKo) [reply src/features/Profile/components/EditProfile/CompanyProfile.tsx:31]: 멘토님 말씀 보고 다시 생각해보니 placeholder 보다는 value로 처리하는게 맞는것 같습니다..
글자수도 세야하니...

제가 놓칠뻔했던 내용 코멘트 주셔서 감사합니다 !

File: src/features/Profile/hooks/useProfileMutation.ts (hunk 6 above)
Comment (SebellKo) [reply src/features/Profile/hooks/useProfileMutation.ts:9]: 넵 ! 확인 했습니다 !

File: src/mocks/profile.ts (hunk 8 above)
Comment (SebellKo) [reply src/mocks/profile.ts:32]: 넵... 😂😂

File: src/features/Profile/hooks/useProfileQuery.ts (hunk 7 above)
Comment (SebellKo) [reply src/features/Profile/hooks/useProfileQuery.ts:5]: 다시 수정해 보도록 하겠습니다 ..!!!

**Pull request merged**